pkg/kubelet/container/container_gc.go

```diff
@@ -86,3 +88,14 @@ func (cgc *realContainerGC) DeleteAllUnusedContainers(ctx context.Context) error
 	klog.InfoS("Attempting to delete unused containers")
 	return cgc.runtime.GarbageCollect(ctx, cgc.policy, cgc.sourcesReadyProvider.AllReady(), true)
 }
+
+func (cgc *realContainerGC) IsWriteableLayerSeparateFromReadOnlyLayer(ctx context.Context) bool {
+	resp, err := cgc.runtime.ImageFsInfoV2(ctx)
+	if err != nil {
+		return false
+	}
+	if resp.ContainerFilesystems == nil || resp.ImageFilesystems == nil {
+		return false
+	}
+	return resp.ContainerFilesystems[0].FsId.Mountpoint != resp.ImageFilesystems[0].FsId.Mountpoint
```

Comment:
you need to do a len check too. it's possible the slice is not nil but the length is 0

Reply:
Good call.  I added a test case for that and another one in case FsId is not specified.